File: frontend/src/types/card.ts
export interface Card {
  id: number
  name: string
  type: string
  frameType: string
  description: string
  level: number | null
  atk: number | null
  def: number | null
  race: string
  attribute: string
  archetype: string
  imageUrl?: string | null
}

File: backend/models/card.py
from dataclasses import dataclass


@dataclass
class Card:
    id: int
    name: str
    type: str
    frameType: str
    description: str
    level: int | None = None
    atk: int | None = None
    def_: int | None = None
    race: str = ""
    attribute: str = ""
    archetype: str = ""
